service/share/light_availability.go

```diff
@@ -13,24 +13,25 @@ import (
 // DefaultSampleAmount sets the default amount of samples to be sampled from the network by lightAvailability.
 var DefaultSampleAmount = 16
 
-// lightAvailability implements Availability using Data Availability Sampling technique.
+// LightAvailability implements Availability using Data Availability Sampling technique.
 // It is light because it does not require the downloading of all the data to verify
 // its availability. It is assumed that there are a lot of lightAvailability instances
 // on the network doing sampling over the same Root to collectively verify its availability.
-type lightAvailability struct {
+type LightAvailability struct {
 	bserv blockservice.BlockService
 }
 
 // NewLightAvailability creates a new light Availability.
-func NewLightAvailability(bserv blockservice.BlockService) Availability {
-	return &lightAvailability{
+func NewLightAvailability(bserv blockservice.BlockService) *LightAvailability {
+	la := &LightAvailability{
 		bserv: bserv,
 	}
+	return la
 }
 
 // SharesAvailable randomly samples DefaultSamples amount of Shares committed to the given Root.
 // This way SharesAvailable subjectively verifies that Shares are available.
-func (la *lightAvailability) SharesAvailable(ctx context.Context, dah *Root) error {
+func (la *LightAvailability) SharesAvailable(ctx context.Context, dah *Root) error {
```

Comment:
why export? Same question for full avail?

Reply:
forgot to push commit with reverting these changes.